storage/localstore/mode_set.go

```diff
@@ -393,3 +399,31 @@ func (db *DB) setUnpin(batch *leveldb.Batch, addr chunk.Address) (err error) {
 
 	return nil
 }
+
+// setReUpload adds a pinned chunk to the push index so that it can be re-uploaded to the network
+func (db *DB) setReUpload(batch *leveldb.Batch, addr chunk.Address) (err error) {
+	item := addressToItem(addr)
```

Comment:
why is this not enough?

```golang
return db.pushIndex.PutInBatch(batch, retrievalDataIndexItem)
```

Reply:
this code takes place in line `425`, but before that, we check that the chunk is pinned (for this we call `addressToItem` first) and also not _already_ in the push index (line `420`)

Reply:
That is correct. We have to ensure that the item is not already in the push index, not to overwrite the tags, to get the store timestamp value and to ensure that the chunk is pinned to avoid gc on it while waiting its queue to be pushed.